Beautify with changes

AaronVerDow opened the pull request:
This is a copy of https://github.com/openscad/openscad/pull/6084 with the formatting changes applied.

https://github.com/openscad/openscad/issues/6055

kintel commented:
I'll add some exceptions using `// clang-format off` comments, especially for formatted matrix initializations and a few other places where clang-format didn't do well.

Will just push to your branch unless you have other preferences.

kintel commented:
One thing to mention: Different versions of clang-format may yield different formatting results. I had clang-format 19 installed, but the GitHub action seems to use clang-format 18. I updated to clang-format 20, which seems to give the same results as v18...

AaronVerDow commented:
> One thing to mention: Different versions of clang-format may yield different formatting results. I had clang-format 19 installed, but the GitHub action seems to use clang-format 18. I updated to clang-format 20, which seems to give the same results as v18...

I made this clearer in the docs and added some notes on the different versions.
